subsys/bluetooth/controller/ll_sw/ull.c

```diff
@@ -2431,17 +2439,20 @@ static inline int rx_demux_rx(memq_link_t *link, struct node_rx_hdr *rx)
 		ull_sync_established_report(link, rx);
 	}
 	break;
-#if defined(CONFIG_BT_CTLR_DF_SCAN_CTE_RX)
-	case NODE_RX_TYPE_IQ_SAMPLE_REPORT: {
+#endif /* CONFIG_BT_CTLR_SYNC_PERIODIC */
+#endif /* CONFIG_BT_CTLR_ADV_EXT */
+#endif /* CONFIG_BT_OBSERVER */
+
+#if defined(CONFIG_BT_CTLR_DF_SCAN_CTE_RX) || defined(CONFIG_BT_CTRL_DF_CONN_CTE_RX)
+	case NODE_RX_TYPE_SYNC_IQ_SAMPLE_REPORT:
```

Comment:
Added `__fallthrough;`

Reply:
As there is no specific code fallthrough, it is not necessary. I would exceptionally add (when there is no code) when checkpatch script misundertands due to compilation compilations of the `case`.

Reply:
I haven't read about fallthrough before, hence I assumed it as a good practice e.g. in host. 
After some reading, I agree with @cvinayak here is is not necessary. There is no code executed for `case NODE_RX_TYPE_SYNC_IQ_SAMPLE_REPORT`, hence it is not necessary to use the statement.

BTW. Since C23 release the `[[ fallthrough ]]` or `[[ __fallthrough__ ]]` attribute will be part of standard.

Reply:
Removed `__fallthrough` from places where it was added.

Reply:
Looks like @cvinayak is correct that if there is no code before the fallthrough, this can be omitted. From searching in the code base, it does look like `__fallthrough` is still used many places without any code, but it does not seem to be something required by the Linux kernel style (which we also follow).